mountpoint-s3/src/fstab.rs

```diff
@@ -64,36 +69,66 @@ impl FsTabCliArgs {
         }
     }
 
-    fn validate_options(options: &[String]) -> Result<(), clap::Error> {
+    fn validate_options(options: &[String], prefix_allowed: bool) -> Result<(), clap::Error> {
         if options.iter().any(|x| x.starts_with("-")) {
-            return Err(Command::error(
-                &mut Self::command(),
+            return Err(Self::command_error(
                 ErrorKind::InvalidValue,
                 "Cannot prefix arguments with `--` when using fstab style arguments.",
             ));
         }
         if options.iter().any(|x| x == "ro") && options.iter().any(|x| x == "rw") {
-            return Err(Command::error(
-                &mut Self::command(),
+            return Err(Self::command_error(
                 ErrorKind::ArgumentConflict,
                 "Cannot use 'ro' flag combined with 'rw' flag.",
             ));
         }
-        if let Some(disallowed_option) = options
-            .iter()
-            .find(|x| ["foreground", "read-only"].contains(&x.as_str()))
-        {
-            return Err(Command::error(
-                &mut Self::command(),
+        if options.iter().any(|x| x == "foreground") {
+            return Err(Self::command_error(
+                ErrorKind::InvalidValue,
+                "Cannot use 'foreground' flag when using fstab style arguments.",
+            ));
+        }
+        if options.iter().any(|x| x == "read-only") {
+            return Err(Self::command_error(
                 ErrorKind::InvalidValue,
-                format!(
-                    "Cannot use '{}' flag when using fstab style arguments.",
-                    disallowed_option
-                ),
+                "Cannot use 'read-only' flag when using fstab style arguments. Use 'ro' instead",
+            ));
+        }
+        if !prefix_allowed && options.iter().any(|x| x.starts_with("prefix")) {
+            return Err(Self::command_error(
+                ErrorKind::ArgumentConflict,
+                "Cannot use 'prefix' flag when using s3:// URIs with a prefix",
             ));
         }
         Ok(())
     }
+
+    fn command_error(error_kind: ErrorKind, message: impl fmt::Display) -> clap::Error {
+        Command::error(&mut Self::command(), error_kind, message)
+    }
+
+    fn parse_bucket_prefix(&self) -> Result<(&str, Option<&str>), clap::Error> {
```

Comment:
It's not great that we split the logic between this method and the `parse_bucket_name` function, besides some code duplication.
Since we plan to move `CliArgs` into this crate soon and we are also considering allowing the "s3://" syntax outside of fstab, shall we pause this review and see if we can combine the 2 pieces of code after the move?